zenoh/src/net/runtime/orchestrator.rs

```diff
@@ -87,25 +85,73 @@ impl Runtime {
                     bail!("No peer specified and multicast scouting desactivated!")
                 }
             }
-            _ => {
-                for locator in &peers {
-                    match tokio::time::timeout(
-                        CONNECTION_TIMEOUT,
-                        self.manager().open_transport_unicast(locator.clone()),
-                    )
-                    .await
-                    {
-                        Ok(Ok(_)) => return Ok(()),
-                        Ok(Err(e)) => log::warn!("Unable to connect to {}! {}", locator, e),
-                        Err(e) => log::warn!("Unable to connect to {}! {}", locator, e),
-                    }
+            _ => self.connect_peers(&peers).await,
```

Comment:
Clients cannot use the same `connect_peers` function as peers. Clients by definition can only be connected to at most one remote process at any point in time.

Reply:
Splitted connect_peers_impl to two methods, one for single_link and other for multiply_links